packages/react/src/primitives/Menu/__tests__/menu.test.tsx

```diff
@@ -116,6 +115,131 @@ describe('Menu', () => {
+      render(
+        <Menu>
+          <MenuItem>Option 1</MenuItem>
+          <MenuItem>Option 2</MenuItem>
+          <MenuItem>Option 3</MenuItem>
+        </Menu>
+      );
+
+      await waitFor(async () => {
+        const menuButton = await screen.findByTestId(MENU_TRIGGER_TEST_ID);
+
+        menuButton.dispatchEvent(
+          new KeyboardEvent('keydown', {
+            bubbles: true,
+            key: 'Enter',
+          })
+        );
+
+        expect(menuButton).toHaveAttribute('data-state', 'open');
+        expect(screen.queryByText('Option 1')).toBeVisible();
+
+        const openMenuButton = await screen.findByTestId(MENU_TRIGGER_TEST_ID);
+        openMenuButton.dispatchEvent(
+          new KeyboardEvent('keydown', {
+            bubbles: true,
+            key: 'Enter',
+          })
+        );
+
+        expect(openMenuButton).toHaveAttribute('data-state', 'closed');
+        expect(screen.queryByText('Option 1')).toBeNull();
+      });
+    });
+
+    it('should disable the trigger with `disabled` prop', async () => {
+      render(
+        <div style={{ pointerEvents: 'auto' }}>
+          <Menu trigger={<button disabled={true} id={MENU_TRIGGER_TEST_ID} />}>
+            <MenuItem>Option 1</MenuItem>
+            <MenuItem>Option 2</MenuItem>
+            <MenuItem>Option 3</MenuItem>
+          </Menu>
+        </div>
+      );
+
+      await waitFor(async () => {
+        const disabled = await screen.findByRole('button');
+        // Native html elements should only have the `disabled` attribute
+        expect(disabled).toHaveAttribute('disabled');
+
+        disabled.dispatchEvent(
+          new KeyboardEvent('keydown', {
+            bubbles: true,
+            key: 'Enter',
+          })
+        );
+        expect(disabled).toHaveAttribute('data-state', 'closed');
+        expect(screen.queryByText('Option 1')).toBeNull();
+      });
+    });
+
+    it('should disable the trigger with `isDisabled` prop', async () => {
+      render(
+        <div style={{ pointerEvents: 'auto' }}>
+          <Menu
+            trigger={
+              <MenuButton isDisabled={true} testId={MENU_TRIGGER_TEST_ID} />
+            }
+          >
+            <MenuItem>Option 1</MenuItem>
+            <MenuItem>Option 2</MenuItem>
+            <MenuItem>Option 3</MenuItem>
+          </Menu>
+        </div>
+      );
+      await waitFor(async () => {
+        const disabled = await screen.findByTestId(MENU_TRIGGER_TEST_ID);
+        // Amplify UI components should have the disabled class
+        expect(disabled).toHaveClass('amplify-button--disabled');
+        expect(disabled).toHaveAttribute('disabled');
+
+        disabled.dispatchEvent(
+          new KeyboardEvent('keydown', {
+            bubbles: true,
+            key: 'Enter',
+          })
+        );
+        expect(disabled).toHaveAttribute('data-state', 'closed');
+        expect(screen.queryByText('Option 1')).toBeNull();
```

Comment:
I'll split this up to be more modular but this test is checking that `isDisabled` "works" by applying styles and preventing events, because it was doing neither before.

This same keyboard event opens a menu when it's not disabled, which is validated in the previous test. I thought it was safe to assume that it would have the same behavior here otherwise the previous test would be failing. The disabled state should force the button to remain in the default state so triggering an event which would otherwise change the state seems like the only way to assert the events aren't being propagated while the item is disabled. Open to suggestions but firing an event and asserting nothing happens seems to be the only way to test the behavior to prevent this bug in the future.

Reply:
To be clear, don't think that we need to assert for behavior of the `MenuButton` here. That should be handled in testing specific to `MenuButton`, if there are no unit tests currently covering disabled state and styling please add one. 

In regards to validation of whether or not the disabled state of the `trigger` element is applied to the Radix `DropdownMenuTrigger` we should just be able to target that element and inspect it, no need for click events.

Think the test you want for asserting whether or not the `Menu` behaves as expected with between endable/disabled states requires toggling the `isDisabled` prop of the `trigger` element between `true` and `false`, and using click events to create a baseline that provides certainty that the events have an actual impact on the assertions